File: programmers/82612.py
import unittest


class Solution:
    def insufficient_amount(self, price, money, count):
        required_amount = sum(map(lambda x: x * price, range(1, count + 1)))
        return required_amount - money if required_amount > money else 0


class Test(unittest.TestCase):
    def test_solution(self):
        solution = Solution()
        self.assertEqual(solution.insufficient_amount(3, 20, 4), 10)


if __name__ == "__main__":
    unittest.main()

File: programmers/82612.go
package programmers

func Abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func InsufficientAmount(price, money, count int) int {
    requiredAmount := 0
    for i := 1; i < count + 1; i++ {
    	requiredAmount += i * price
    }
    return Abs(min(money - requiredAmount, 0))
}
